lib/lightstep/tracer/transport/http_json.rb

```diff
@@ -0,0 +1,108 @@
+require 'net/http'
+require 'lightstep/tracer/transport/base'
+
+module LightStep
+  module Transport
+    # HTTPJSON is a transport that sends reports via HTTP in JSON format.
+    # It is thread-safe, however it is *not* fork-safe. When forking, all items
+    # in the queue will be copied and sent in duplicate.
+    #
+    # When forking, you should first `disable` the tracer, then `enable` it from
+    # within the fork (and in the parent post-fork). See
+    # `examples/fork_children/main.rb` for an example.
+    class HTTPJSON < Base
+      LIGHTSTEP_HOST = "collector.lightstep.com"
+      LIGHTSTEP_PORT = 443
+      QUEUE_SIZE = 16
+
+      class QueueFullError < LightStep::Error; end
+
+      # Initialize the transport
+      # @param host [String] host of the domain to the endpoind to push data
+      # @param port [Numeric] port on which to connect
+      # @param verbose [Numeric] verbosity level. Right now 0-3 are supported
+      # @param secure [Boolean]
```

Comment:
we've been using this convention lately: https://github.com/lightstep/lightstep-tracer-python/blob/master/lightstep/tracer.py#L28

(not arguing that it's "better", but did want to be consistent)